pulser/sequence.py

```diff
@@ -315,16 +316,17 @@ def declare_channel(self, name: str, channel_id: str,
                 initial_target = None
             else:
                 # "_target" call is not saved
-                self._target(initial_target, name)
+                self._target(cast(Union[Iterable[QubitId],
```

Comment:
Hum... This casting doesn't work with a parametrized `Iterable` imported from `collections.abc`. I you remove the `[QubitId]`, does mypy complain?